libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/permission/PermissionsLauncherFragment.kt

```diff
@@ -0,0 +1,65 @@
+package com.mapbox.navigation.dropin.permission
+
+import android.content.Context
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import androidx.activity.result.ActivityResultCallback
+import androidx.activity.result.ActivityResultCaller
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts.RequestMultiplePermissions
+import androidx.core.content.ContextCompat.checkSelfPermission
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+
+/**
+ * A view-less fragment that requests permissions using [ActivityResultCaller] interface.
+ */
+internal class PermissionsLauncherFragment(
+    private val permissions: Array<String>,
+    private val onResult: ActivityResultCallback<Map<String, Boolean>>
+) : Fragment() {
+
+    private var launcher: ActivityResultLauncher<Array<String>>? = null
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        val areGranted = permissions.fold(false) { acc, permission ->
+            acc && checkSelfPermission(context, permission) == PERMISSION_GRANTED
+        }
+        if (!areGranted) {
+            launcher = registerForActivityResult(RequestMultiplePermissions(), onResult)
+            launcher?.launch(permissions)
```

Comment:
Why do you request all permissions when some of them may already have been granted?

Reply:
That's a great suggestion. We could just request the ones that are missing.

Reply:
Updated. Thx.